src/lib.ts

```diff
@@ -23,18 +23,22 @@ export async function getInternalLinks(
 
   const analyzedLinks: AnalyzedLink[] = [];
 
-  // I won't worry about using Promise.all with some kind of batching here, maybe refactor later.
-  for (const nodePath of nodePaths) {
-    // eslint-disable-next-line no-await-in-loop
-    const contents = await fs.readFile(nodePath, 'utf-8');
+  const promiseList = nodePaths.map((nodePath) => fs.readFile(nodePath, 'utf-8'));
+  const contentList: string[] = [];
+  
+  await Promise.all(promiseList)
+  .then(result => result.forEach(i => contentList.push(i)))
+  .catch(console.log);
```

Comment:
I'd not use `catch` here since it essentially would prevent the exception unwinding in this case. We want to stop the code execution in case of an error, no?